### PR TITLE
Fix for edge case rounding errors breaking bad triangle selection

### PR DESCRIPTION
**Pull request by Spriteanon**
After some testing, discovered that in some edge cases "Triangle.is_point_inside_circumcircle(point)" will flag the corners as being inside the circumcircle (which of course is counter to the logic of the algorithm). By simply checking and ignoring its own corners, all of my personal issues relating to broken polygons similar to #1 have been resolved.

### CODE REVIEW

**Comment by bartekd97** [comment]
Ah, I see whats going on here.
Can you check if you get same result by simply changing '<" operator with "<="?
This would be a bit cheaper than extra comparison statements.

**Comment by Spriteanon** [comment]
I just checked, but as expected, that makes things worse. The issue is that we want a "false" answer if the triangle is looking at one of its own corners, so if anything we want to check to be less permissive.

The only other solution I can think of is passing the point to the _find_bad_triangle() function and having it ignore triangles that the point is a part of, but the way I'm counting that would essentially result in the same number of checks anyway.

It might be worth experimenting with a multiplier for radius_sqr in the high 0.9x-s. That would be imprecise, and could have a cascade effect, but that's the only other thing I can think of that doesn't involve checking the corners of triangles.



Also, before you wrote I was in the process of writing up further clarification of what the bug's nature was, it might give you some ideas:

Basically the problem resulted in new "triangles" getting created where it was actually just a line segment, with the point matching one of the ends of the edge in the following segment of triangulate(), since the edge came from a triangle that the point was a part of:
```
for edge in polygon:
	triangulation.append(Triangle.new(point, edge.a, edge.b))
```
This led to among other things an obvious division by 0 while trying to calculate circum, and by extension any attempt to generate voronoi from the centres of the circumference.

**Comment by bartekd97** [comment]
Okay, got it.
Multiplier like 0.99x won't wok.
The potential solution would be comparing with epsilon, e.g. ```center.distance_squared_to(point) < (radius_sqr - 0.00001)```.
If you can confirm epsilon solution, then it would be best solution, but its value has to be kept as low as possible.

Otherwise we will just stay with your solution.

**Comment by Spriteanon** [comment]
Did some experimenting with different values. In my tests the problem didn't go away until I increased the correction to `-0.08`, and the increments up there had a lot of minor improvements, which to me suggests with different point sets, it's likely to crop up again if we go about things like this.

One thing that doesn't exactly sit well with me, is that during these incremental tests I kept printing out for myself instances of triangle being generated with 2 identical corners, (at least, where `point` was equal to either ends of an edge, and at a certain point those errors stopped, but not all broken polygons disappeared, so there seems to be something more to this than what I originally reasoned out. Sadly, I'm out of ideas as to what might be going on there, especially since, again, the problem is resolved by just checking the corners so no idea what's causing it.

In any case, personally I'd stick with the most reliable method of simply checking the corners. Either that, or include two versions, with people being able to switch between the higher performance version with possible issues, and the slightly slower but more accurate method.

As a final note, if we do go with my previous solution, I since realized a more elegant version would be to instead of writing out the 3 checks, the code could simply call `is_corner(point)`. It should be functionally the same in terms of performance, but looks nice since I don't know how I missed the function literally just beneath the one I adjusted.

**Comment by bartekd97** [comment]
I will merge this as is then.
Separating this function looks more elegant, but I wonder that it will cause unnecessary function calls on stack, so I'd stay with your original solution.
Thanks.
